internal/setup/setup.go

```diff
@@ -236,6 +243,20 @@ func SetupWith(ctx context.Context, config interface{}, l envconfig.Lookuper) (*
 
 			logger.Infow("authorizedapp", "config", aaConfig)
 		}
+
+		// Start the JWKS key downloader.
+		if provider, ok := config.(JWKSManagerConfigProvider); ok {
+			logger.Info("configuring jwks manager")
+
+			jwksConfig := provider.JWKSManagerConfig()
+			jwks, err := jwks.NewFromEnv(ctx, db, jwksConfig)
+			if err != nil {
+				// Ensure the db is closed on an error.
+				defer db.Close(ctx)
+				return nil, fmt.Errorf("unable to start jwks manager: %w", err)
+			}
+			jwks.Start()
```

Comment:
Does it make sense for this to return an error? Looks like it just spawns a goroutine.

Reply:
background refresh may or may not work on serverless infrastructure due to CPU throttling when not serving requests.

Also - does this need to be started in all servers? would it make sense to have this refresh be a periodic job that that syncs keys into the database and then rely on existing public key caching infrastructure?

Reply:
obsolete.